src/main/java/org/jabref/gui/bibtexextractor/BibtexExtractorViewModel.java

```diff
@@ -1,32 +1,99 @@
 
 public class BibtexExtractorViewModel {
 
     private final StringProperty inputTextProperty = new SimpleStringProperty("");
     private final BibDatabaseContext bibdatabaseContext;
+    private List<BibEntry> extractedEntries;
+    private final BibDatabaseContext newDatabaseContext;
+    private boolean directAdd;
+    private DialogService dialogService;
+    private GrobidCitationFetcher currentCitationfetcher;
 
     public BibtexExtractorViewModel(BibDatabaseContext bibdatabaseContext) {
         this.bibdatabaseContext = bibdatabaseContext;
+        newDatabaseContext = new BibDatabaseContext(new Defaults(BibDatabaseMode.BIBTEX));
+        dialogService = JabRefGUI.getMainFrame().getDialogService();
     }
 
     public StringProperty inputTextProperty() {
         return this.inputTextProperty;
     }
 
-    public void startExtraction() {
+    public void startParsing(boolean directAdd) {
+        this.directAdd = directAdd;
+        this.extractedEntries = null;
+        Task<Void> parseUsingGrobid = new Task<Void>() {
+            @Override
+            protected Void call() throws Exception {
+                try {
+                    currentCitationfetcher = new GrobidCitationFetcher(
+                        JabRefPreferences.getInstance().getImportFormatPreferences(),
+                        Globals.getFileUpdateMonitor(),
+                        Globals.prefs
+                    );
+                    extractedEntries = currentCitationfetcher.performSearch(inputTextProperty.getValue());
+                } catch (FetcherException e) {
+                    extractedEntries = new ArrayList<>();
+                }
+                Platform.runLater(() -> executeParse());
+                return null;
+            }
+        };
+        dialogService.showProgressDialogAndWait(Localization.lang("Your text is being parsed.."),Localization.lang( "Please wait while we are parsing your text"), parseUsingGrobid);
+        Globals.TASK_EXECUTOR.execute(parseUsingGrobid);
+    }
 
+    public void executeParse() {
+        if (extractedEntries.size() > 0) {
+            if (directAdd) {
+                BibtexKeyGenerator bibtexKeyGenerator = new BibtexKeyGenerator(newDatabaseContext, Globals.prefs.getBibtexKeyPatternPreferences());
+                for (BibEntry bibEntries: extractedEntries) {
+                    bibtexKeyGenerator.generateAndSetKey(bibEntries);
+                    newDatabaseContext.getDatabase().insertEntry(bibEntries);
+                }
+                newDatabaseContext.setMode(BibDatabaseMode.BIBTEX);
+                JabRefGUI.getMainFrame().addTab(newDatabaseContext,true);
+            } else {
+                for (BibEntry bibEntry: extractedEntries) {
+                    this.bibdatabaseContext.getDatabase().insertEntry(bibEntry);
+                    JabRefGUI.getMainFrame().getCurrentBasePanel().showAndEdit(bibEntry);
+                    trackNewEntry(StandardEntryType.Article);
```

Comment:
The track event also needs to be changed I think.

Reply:
For the track event (it's a method someone else in JabRef wrote) what specifically should we change?

Reply:
The event name "NewEntry" is reserved for the case when the user creates a new empty entry. I think it is sufficient to track once that the user parsed something with grobid (i.e. `Globals.getTelemetryClient().ifPresent(client -> client.trackEvent("ParseWithGrobid", new HashMap<>(), new HashMap<>()));`

Reply:
Fixed here: deb65c000c8f6b1ff1b362d265f5fedfec721671